### ISSUE TITLE
[🐛 Bug]: Async methods cannot be easily recognized

### ISSUE THREAD

**Issue by nvborisenko** [SeleniumHQ/selenium#12671]
### What happened?

We cannot easily understand whether a method is asynchronous or not. When we write code, we use synchronous methods and asynchronous beside. There is high chance to make a mistake when dealing with async methods, and don't await them.

Example:
```csharp
driver.Network.StartMonitoring();
```

Who knows it's async or not. Tooling (like IDE) doesn't help us (at least for me) to understand, there is even no warnings.

Even Jim made a mistake:
![image](https://github.com/SeleniumHQ/selenium/assets/22616990/a77d8045-6e9e-4d72-87eb-a7220199c3d5)

I also went through open issues in this repository and observed that people don't await `StartMonitoring()` method.

# Solution
Add `Async` to all async methods according [naming convention](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/task-asynchronous-programming-model#BKMK_NamingConvention).

> By convention, methods that return commonly awaitable types (for example, Task, Task<T>, ValueTask, ValueTask<T>) should have names that end with "Async".

### How can we reproduce the issue?

```shell
driver.Network.StartMonitoring();
```

### Relevant log output

```shell
no output
```

### Operating System

any

### Selenium version

any dotnet, selenium 4.0+

### What are the browser(s) and version(s) where you see this issue?

any

### What are the browser driver(s) and version(s) where you see this issue?

any

**Comment by rohitkumarWUSTL** [comment]
I'm looking to be a first-time contributor; would this be a good issue for me to pick up?

**Comment by titusfortner** [comment]
Well, this is going to be a huge one and likely won't be released until Selenium 5, and we don't even know what that is going to look like, yet. But @nvborisenko can probably help you find something if you want to work in .NET code.

**Comment by titusfortner** [comment]
It's easier to chat about it on Slack — https://www.selenium.dev/support/#ChatRoom

**Comment by rohitkumarWUSTL** [comment]
Ah, those would be good reasons for me to not pick this up as a first issue in that case 😅 

I also missed that this issue was re: .NET - I'm primarily looking to contribute in Java, if possible - but regardless I'll join the Slack and catch up. Thank you!

**Comment by MgSam** [comment]
I'm interested in seeing what the async methods being added to Selenium will look like. Is there a specific branch to check out?

**Comment by nvborisenko** [comment]
When we think about async methods naming, can I raise here one more issue about "async properties"? It could be separate issue for discussion... anyway: example: https://github.com/SeleniumHQ/selenium/issues/13629

Summary: properties should represent a state, calculated properties should not be "sync-over-async".

CC: @jimevans

**Comment by titusfortner** [comment]
I seem to recall @jimevans saying something about how he wouldn't rely on properties in an async library. Jim, did I get that right, or did I misunderstand what you were saying?

**Comment by Piedone** [comment]
Is this issue also about introducing more asynchronous methods where Selenium currently only has sync APIs?

**Comment by nvborisenko** [comment]
No, it is only about renaming `Task DoSomething()` to `Task DoSomethingAsync()`. Your point is valid, I don't think it will be in place in near feature.

**Comment by Piedone** [comment]
I see, thanks.

**Comment by nvborisenko** [comment]
One more user doesn't await Task - https://github.com/SeleniumHQ/selenium/issues/17350

**Comment by Piedone** [comment]
FWIW, there are code analyzers that will shout if you don't await an async call. This issue is still valid, of course, and renaming async methods would improve DX, just adding that this doesn't affect everyone to the same level.
